sfs: adapt tests to sqlite WAL mode

irq0 opened the pull request:
Since the SQLite WAL PR, there is now more then one database file. This makes the tests fail, as they assume a single database file.

Signed-off-by: Marcel Lauhoff <marcel.lauhoff@suse.com>

Commit:
tests/sfs: separate data file count and db file exists check

Since the SQLite WAL PR, there is now more then one database file.
This makes the tests fail, as they assume a single database file.

Fix this by having separate tests for data file count and database
file existence. This makes tests independent on the number of SQLite
database files.

Signed-off-by: Marcel Lauhoff <marcel.lauhoff@suse.com>

jecluis reviewed (dismissed):
lgtm - should the tests pass, good to merge.

Commit:
sfs: copy sqlite wal file during metadata compat check

Copy the wal file if it exists. It does if writes happend to the db.
It may not exists during unit tests or on fresh databases.

Signed-off-by: Marcel Lauhoff <marcel.lauhoff@suse.com>

Commit:
tests/sfs: set up TestDB with sqlite WAL mode

Set up sqlite in TestDBs like we do in the SFS DBConn.

Signed-off-by: Marcel Lauhoff <marcel.lauhoff@suse.com>

giubacc reviewed (approved):
lgtm
